pkg/apis/authorization/types.go

```diff
@@ -17,13 +17,19 @@ limitations under the License.
 package authorization
 
 import (
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
+// +genclient=true
+// +nonNamespaced=true
+// +noMethods=true
+
 // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
 // spec.namespace means "in all namespaces".
 type SubjectAccessReview struct {
 	unversioned.TypeMeta
+	api.ObjectMeta
```

Comment:
> What is kubectl create doing that it requires there to be ObjectMeta?
> What would be the minimal changes to kubectl to be able to handle these objects without ObjectMeta?

It seems to be required API server side for setting the self-link.  For a long time we had a special-cased hack in origin to skip that for certain types.  I don't know how its working downstream now.

Given the difficulty in making it work and the amount of code that assumes that every type has `ObjectMeta`, this seems lower friction.